gazelle_setup.bzl

```diff
@@ -0,0 +1,24 @@
+# Copyright 2019 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Dependency registration helpers for the gazelle plugin."""
+
+load("@bazel_gazelle//:deps.bzl", "gazelle_dependencies")
+load("@io_bazel_rules_go//go:deps.bzl", "go_register_toolchains", "go_rules_dependencies")
+
+def bazel_skylib_gazelle_plugin_setup():
+    go_rules_dependencies()
+    go_register_toolchains(version = "1.17.1")
```

Comment:
Have you tried calling this macro from a `WORKSPACE` that already sets up a Go toolchain somewhere else? You might have to make running this configurable via a parameter.

If possible, also bump the version to `1.18.7` as `1.17` is out of support. `1.19` is available, but requires a relatively recent version of `rules_go`.

Reply:
Done and done.